### PR TITLE
Use boost dynamic_bitset in segcore

### LINKED ISSUE #16475: [Enhancement]: Use boost dynamic_bitset in segcore
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

Currently ConcurrentBitset is used in deleting records and BitsetType(boost::dynamic_bitset) in scalar filtering. After testing, it is found that boost::dynamic_bitset has the same performance and richer interfaces. Therefore, segcore uniformly uses BitsetType.

### Why is this needed?

Reduce code maintenance complexity

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by czs007**
fix: #16475 
Signed-off-by: zhenshan.cao <zhenshan.cao@zilliz.com>

### CODE REVIEW

**Review by cydrain — COMMENTED** [review]
can move files under core/thirdparty/boost_ext to common ?

File: internal/core/src/common/BitsetView.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#pragma once
+
+#include <deque>
+#include <boost_ext/dynamic_bitset_ext.hpp>
+#include "common/Types.h"
+#include "knowhere/utils/BitsetView.h"
+
+namespace milvus {
+
+using BitsetSimple = std::deque<BitsetType>;
```
[the remark this answers is not in the document]
Comment (czs007) [reply internal/core/src/common/BitsetView.h:26]: it is not  used by now. perfer to delete it in the next pr.

File: internal/core/src/common/BitsetView.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#include <boost_ext/dynamic_bitset_ext.hpp>
+#include "common/Types.h"
+#include "common/BitsetView.h"
+#include "utils/Utils.h"
+
+namespace milvus {
+
+// negate bitset, and merge them into one
+aligned_vector<uint8_t>
+AssembleNegBitset(const BitsetSimple& bitset_simple) {
```
Comment (cydrain) [inline internal/core/src/common/BitsetView.cpp:26]: This API should be AssembleBitset or BitsetAssemble.
To flip all bit in bitset, we can use boost_bitset::flip().

File: internal/core/src/common/Types.h [hunk 3]
```diff
@@ -74,17 +73,8 @@ using FieldName = fluent::NamedType<std::string, impl::FieldNameTag, fluent::Com
 using FieldOffset = fluent::NamedType<int64_t, impl::FieldOffsetTag, fluent::Comparable, fluent::Hashable>;
 using SegOffset = fluent::NamedType<int64_t, impl::SegOffsetTag, fluent::Arithmetic>;
 
-using BitsetView = faiss::BitsetView;
-inline BitsetView
-BitsetSubView(const BitsetView& view, int64_t offset, int64_t size) {
-    if (view.empty()) {
-        return BitsetView();
-    }
-    assert(offset % 8 == 0);
```
Comment (cydrain) [inline internal/core/src/common/Types.h:83]: better assert(offset % 8 == 0 and offset < view.size())

File: internal/core/src/common/BitsetView.cpp (hunk 2 above)
Comment (czs007) [reply internal/core/src/common/BitsetView.cpp:26]: This function is not been used ethier.

File: internal/core/src/common/BitsetView.cpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "common/BitsetView.h"
+
+namespace milvus {
+
+BitsetView
+BitsetView::subview(size_t offset, size_t size) const {
```
Comment (cydrain) [inline internal/core/src/common/BitsetView.cpp:22]: Since all implementations are coded in BitsetView.h, suggest to move subview to BitsetView.h

Comment (czs007) [reply internal/core/src/common/BitsetView.cpp:22]: sounds good. Prefer to do this in next pr.

File: internal/core/src/common/BitsetView.cpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    if (empty()) {
+        return BitsetView();
+    }
+    assert(offset % 8 == 0);
+    assert((offset + size) <= this.size());
```
Comment (czs007) [inline internal/core/src/common/BitsetView.cpp:27]: @jennyli-z

**Pull request merged**